app/services/mobilize.py

```diff
@@ -46,36 +44,49 @@ def database_error(context: str) -> JSONResponse:
     return JSONResponse(status_code=502, content={"detail": message})
 
 
-@mobilize.get("/uploadcsv", response_class=HTMLResponse)
+@mobilize.get("/uploadcsv", response_class=FileResponse)
 async def upload_csv(request: Request):
-    return templates.TemplateResponse("upload_csv.html", {"request": request})
+    return FileResponse("docs/upload_csv.html")
 
 
-@mobilize.post("/transfercsv")
-async def transfer_csv(file: UploadFile = File(...), force_transfer: bool = False):
+@mobilize.post("/transfercsv", response_class=HTMLResponse)
+async def transfer_csv(
+    request: Request, file: UploadFile = File(...), force_transfer: bool = False
+):
     if file.filename.endswith(".csv"):
-        df = pd.read_csv(file.file, na_filter=False)
-        df = df.astype(str)
+        try:
+            df = pd.read_csv(file.file, na_filter=False)
+        except:
+            return templates.TemplateResponse(
+                "error.html", {"request": request, "msg": "Error reading csv file"}
+            )
```

Comment:
You need to log the error before returning it:
```suggestion
            message = log_error("Error reading csv file")
            return templates.TemplateResponse(
                "error.html", {"request": request, "msg": message}
            )
```